modules/store.js

```diff
@@ -75,6 +75,15 @@ export const reducer = handleActions({
 
 }, initialState);
 
+export const reducer = function wrappedReducer(state, action) {
+  // if state is undefined then it can't be converted to mutable
+  let mutableState = state;
+  if (mutableState !== undefined) {
+    mutableState = getMutableState(state);
+  }
+  return getImmutableState(reduxAsyncReducer(mutableState, action));
+};
+
```

Comment:
Thats fine, thx.
Last change that I'd to like to avoid is renaming old reducer to a new name and setting new reducer as the main one. That would require people with mutable state to read on the new change and explicitly select new naming. That being said - if we reverse the naming of the reducers around we can do a pure feat without any real changes to old code

> On 16 Aug 2016, at 23:31, Todd Bluhm <notifications@github.com> wrote:
> 
> In modules/store.js:
> 
> > @@ -75,6 +75,15 @@ export const reducer = handleActions({
> >  
> >  }, initialState);
> >  
> > +export const reducer = function wrappedReducer(state, action) {
> > +  // if state is undefined then it can't be converted to mutable
> > +  let mutableState = state;
> > +  if (mutableState !== undefined) {
> > +    mutableState = getMutableState(state);
> > +  }
> > +  return getImmutableState(reduxAsyncReducer(mutableState, action));
> > +};
> > +
> Done. Let me know if this new change is good enough
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>